graphql/src/main/java/com/linecorp/armeria/server/graphql/GraphQLServiceBuilder.java

```diff
@@ -0,0 +1,224 @@
+     */
+    public GraphQLServiceBuilder typeVisitors(Iterable<? extends GraphQLTypeVisitor> typeVisitors) {
+        typeVisitorBuilder.addAll(requireNonNull(typeVisitors, "typeVisitors"));
+        return this;
+    }
+
+    /**
+     * Adds the {@link Instrumentation}s.
+     */
+    public GraphQLServiceBuilder instrumentation(Instrumentation... instrumentations) {
+        return instrumentation(ImmutableList.copyOf(requireNonNull(instrumentations, "instrumentations")));
+    }
+
+    /**
+     * Adds the {@link Instrumentation}s.
+     */
+    public GraphQLServiceBuilder instrumentation(Iterable<? extends Instrumentation> instrumentations) {
+        instrumentationBuilder.addAll(requireNonNull(instrumentations, "instrumentations"));
+        return this;
+    }
+
+    /**
+     * Adds the {@link GraphQL.Builder} consumers.
+     */
+    public GraphQLServiceBuilder configure(Consumer<GraphQL.Builder>... configurers) {
+        return configure(ImmutableList.copyOf(requireNonNull(configurers, "configurers")));
+    }
+
+    /**
+     * Adds the {@link GraphQL.Builder} consumers.
+     */
+    public GraphQLServiceBuilder configure(Iterable<Consumer<GraphQL.Builder>> configurers) {
+        configurerBuilder.addAll(requireNonNull(configurers, "configurers"));
+        return this;
+    }
+
+    /**
+     * Creates a {@link GraphQLService}.
+     */
+    public GraphQLService build() {
+        final TypeDefinitionRegistry registry = typeDefinitionRegistry(schemaFileBuilder.build());
+
+        final RuntimeWiring runtimeWiring = buildRuntimeWiring(runtimeWiringConfiguratorBuilder.build());
+        GraphQLSchema schema = new SchemaGenerator().makeExecutableSchema(registry, runtimeWiring);
+        final List<GraphQLTypeVisitor> typeVisitors = typeVisitorBuilder.build();
+        for (GraphQLTypeVisitor typeVisitor : typeVisitors) {
+            schema = SchemaTransformer.transformSchema(schema, typeVisitor);
+        }
+
+        GraphQL.Builder builder = GraphQL.newGraphQL(schema);
+        final List<Instrumentation> instrumentations = instrumentationBuilder.build();
+        if (!instrumentations.isEmpty()) {
+            builder = builder.instrumentation(new ChainedInstrumentation(instrumentations));
+        }
+
+        final List<Consumer<GraphQL.Builder>> configurers = configurerBuilder.build();
+        for (Consumer<GraphQL.Builder> configurer : configurers) {
+            configurer.accept(builder);
+        }
+        return new DefaultGraphQLService(builder.build(), dataLoaderRegistry);
+    }
+
+    private static TypeDefinitionRegistry typeDefinitionRegistry(List<File> schemaFiles) {
+        final TypeDefinitionRegistry registry = new TypeDefinitionRegistry();
+        final SchemaParser parser = new SchemaParser();
+        if (schemaFiles.isEmpty()) {
+            schemaFiles = defaultSchemaFiles();
+        }
+        if (schemaFiles.isEmpty()) {
+            throw new IllegalStateException("Not found schema file(s)");
```

Comment:
nit: Schema file is not found?
How about also logging the applied schema files?